pkg/source/pulsar.go

```diff
@@ -239,7 +249,13 @@ func (p *PulsarConsumerSource) Capture(cp cursor.Checkpoint) (changes chan Chang
 
 func (p *PulsarConsumerSource) Commit(cp cursor.Checkpoint) {
 	if mid, err := pulsar.DeserializeMessageID(cp.Data); err == nil {
```

Comment:
Can't we just inject BatchIdx into `cp.Data` when sending to clients and deserialize it back here?

Reply:
Since we need to use ackTrackers to identify these messages belong to the same batch